pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -1841,242 +1841,345 @@ func getNumberOfPodInterfaces(spec *v1.VirtualMachineInstanceSpec) int {
-				validBus := false
-				for _, b := range buses {
-					if b == bus {
-						validBus = true
-					}
-				}
-				if !validBus {
-					causes = append(causes, metav1.StatusCause{
-						Type:    metav1.CauseTypeFieldValueInvalid,
-						Message: fmt.Sprintf("%s is set with an unrecognized bus %s, must be one of: %v", field.Index(idx).String(), bus, buses),
-						Field:   field.Index(idx).Child(diskType, "bus").String(),
-					})
-				}
+func validateBootOrderValue(field *k8sfield.Path, idx int, disk v1.Disk) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+	if disk.BootOrder != nil && *disk.BootOrder < 1 {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("%s must have a boot order > 0, if supplied", field.Index(idx).String()),
+			Field:   field.Index(idx).Child("bootOrder").String(),
+		})
+	}
+	return causes
+}
 
-				// special case. virtio is incompatible with CD-ROM for q35 machine types
-				if diskType == "cdrom" && bus == v1.DiskBusVirtio {
-					causes = append(causes, metav1.StatusCause{
-						Type:    metav1.CauseTypeFieldValueInvalid,
-						Message: fmt.Sprintf("Bus type %s is invalid for CD-ROM device", bus),
-						Field:   field.Index(idx).Child("cdrom", "bus").String(),
-					})
+func getDiskBus(disk v1.Disk) v1.DiskBus {
+	var diskBus v1.DiskBus
+	if disk.Disk != nil {
+		diskBus = disk.Disk.Bus
+	}
+	if disk.LUN != nil {
+		diskBus = disk.LUN.Bus
+	}
+	if disk.CDRom != nil {
+		diskBus = disk.CDRom.Bus
+	}
+	return diskBus
+}
 
-				}
-				// sata disks (in contrast to sata cdroms) don't support readOnly
-				if disk.Disk != nil && bus == v1.DiskBusSATA && disk.Disk.ReadOnly {
-					causes = append(causes, metav1.StatusCause{
-						Type:    metav1.CauseTypeFieldValueInvalid,
-						Message: fmt.Sprintf("%s hard-disks do not support read-only.", bus),
-						Field:   field.Index(idx).Child("disk", "bus").String(),
-					})
-				}
-			}
+func getDiskType(disk v1.Disk) string {
+	var diskType string
+	if disk.Disk != nil {
+		diskType = "disk"
+	}
+	if disk.LUN != nil {
+		diskType = "lun"
+	}
+	if disk.CDRom != nil {
+		diskType = "cdrom"
+	}
+	return diskType
+}
 
-			// Reject defining DedicatedIOThread to a disk without VirtIO bus since this configuration
-			// is not supported in libvirt.
-			isIOThreadsWithUnsupportedBus := disk.DedicatedIOThread != nil && *disk.DedicatedIOThread &&
-				bus != v1.DiskBusVirtio
-			if isIOThreadsWithUnsupportedBus {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueNotSupported,
-					Message: fmt.Sprintf("IOThreads are not supported for disks on a %s bus", bus),
-					Field:   field.Child("domain", "devices", "disks").Index(idx).String(),
-				})
+func validateBusSupport(field *k8sfield.Path, idx int, disk v1.Disk) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+	bus := getDiskBus(disk)
+	diskType := getDiskType(disk)
+	if len(bus) == 0 {
```

Comment:
Why len? The function returns DiskBus? You could compare it against an empty string